gantry/entrypoint.sh

```diff
@@ -244,16 +220,229 @@ function ensure_pip {
     log_info "Done. Using $(pip --version)"
 }
 
+function run_custom_cmd {
+    local custom_cmd_purpose=$1
+    local custom_cmd=$2
+    if [[ -f "$custom_cmd" ]] && [[ "${custom_cmd: -3}" == ".sh" ]]; then
+        log_info "Sourcing user-defined $custom_cmd_purpose script '$custom_cmd'..."
+
+        # shellcheck disable=SC1090
+        source "$custom_cmd" || return 1
+
+        # Reset shell behavior.
+        set -eo pipefail
+        set +x
+
+        log_info "Done."
+    else
+        log_info "Running user-defined $custom_cmd_purpose command: '$custom_cmd'"
+        eval "$custom_cmd" || return 1
+        log_info "Done."
+    fi
+}
+
+function uv_install_project {
+    local project_file=$1
+    shift 1
+
+    log_info "Installing local project..."
+    capture_logs "uv_pip_install.log" uv pip install "$@" -r "$project_file" --all-extras . || return 1
```

Comment:
I'm conflicted about using `--all-extras` by default. One of the difficulties I had when trying to write a PR for uv support in gantry is that `uv` has a lot of useful flags! I found myself just using the install command a lot. (I'm also a fan of the `uv sync` workflow).

Reply:
As discussed on Slack, I added flags to customize this behavior: c6eead7